### DISCUSSION TITLE
Packaging assets vs CDN

### DISCUSSION THREAD

**Discussion by AshotN** [ampache/ampache#2669] [General]
I want to make this discussion now so I don't forget down the line and the it becomes more work. 

Should we include assets with our client all packaged up, or use a CDN to load them as needed. I am on the side for local, due to privacy concerns.

# Local

## Pros
- Faster local load times
- More privacy friendly
- Greater control (Assets becoming missing for whatever reason in the future, or if they are blocked in some countries)

## Cons
- Slower loading times out of network, due to caching.
- Greater package size

**Comment by usox** [comment]
I'd vote for `local`, too (for privacy reasons) - at least as sane default value if the users should be able to choose. I don't want my mediaplayer to stop working just because some external entity is offline. And I don't want to need external connectivity if ampache is installed in a local network.

**Comment by kuzi-moto** [comment]
Yep, another vote for local. If wanted it could be supplied as an opt-in option.

**Comment by lachlan-00** [comment]
Server releases are full packages so I would probably continue that.
